dotnet/src/Extensions/Planning.StepwisePlanner/Skills/StepwiseStep/SystemPrompt.txt

```diff
@@ -0,0 +1,47 @@
+[INSTRUCTIONS]
+* Answer the user as accurately as possible using the provided functions.
+* Break down the problem step by step and think about the best approach
+* Generate a single [THOUGHT] and an optional single [ACTION]. The resulting output of an [ACTION] will be provided as an [OBSERVATION]
+* Always use the functions provided in the [AVAILABLE FUNCTIONS] section when possible regardless of your internal knowledge or information.
+* Once the necessary observations have been gathered and required actions performed, provide the [FINAL ANSWER] in a clear and human-readable format.
+
+
+[AVAILABLE FUNCTIONS]
```

Comment:
I anecdotally was finding chat model would hallucinate functions quickly until I pulled the available functions out of here and into the first user message before the question. Did you experience anything similar?

Reply:
It did pretty well if I created few shot examples, but I didn't do a lot of experimentation. It certainly seemed to pay more attention to the examples than this definition.